Lesson18/serkan/script.js

```diff
@@ -0,0 +1,182 @@
+  let result = [];
+
+  for (let i = 0; i < array.length; i++) {
+    if (value !== array[i]) {
+      result.push(array[i]);
+    }
+  }
+  return result;
+}
+
+console.log(removeValueFromArray(2, testArrayEx6));
+
+console.log("exercise 6:");
+console.log('----------------------------');
+// 7. Write a function that takes an object and returns the keys as an array.
+// Input: {name: "John", age: 30, city: "New York"}
+// Output: ["name", "age", "city"]
+const testObjEx7 = { name: "John", age: 30, city: "New York" };
+
+function takeObjOfArray (object) {
+  let keys = [];
+  for (let key in object) {
+    keys.push(key);
+  }
+  return keys;
+}
+console.log(takeObjOfArray(testObjEx7));
+
+
+console.log("exercise 7:");
+console.log('----------------------------');
+// 8. Write a function that takes an object and returns the values as an array.
+// Input: {name: "John", age: 30, city: "New York"}
+// Output: ["John", 30, "New York"]
+const testObjEx8 = { name: "John", age: 30, city: "New York" };
+
+function takeValOfArray (value) {
+  let values = [];
+  for (let key in value) {
+    values.push(value[key]);
+  }
+  return values;
+}
+console.log(takeValOfArray(testObjEx8));
+
+
+console.log("exercise 8:");
+console.log('----------------------------');
+// 9. Write a function that takes an object and a key, and returns true if the key exists in the object, otherwise false.
+// Input: ({name: "John", age: 30, city: "New York"}, "age")
+// Output: true
+// Input: ({name: "John", age: 30, city: "New York"}, "surname")
+// Output: false
+const testObjEx9 = { name: "John", age: 30, city: "New York" };
+
+function getKeyArray(object, key) {
+  return key in object;
+}
+
+console.log(getKeyArray(testObjEx9, "name"));
+console.log(getKeyArray(testObjEx9, "surname"));
+
+console.log("Ex 9:");
+
+console.log('----------------------------');
+
+
+// I honestly had a hard time with exercises 10, 11 and 12. I didn't know what to do.
+
+
+
+
+// 10. Write a function that takes an object and a key-value pair, and adds the key-value pair to the object.
+// Input: ({name: "John", age: 30}, "city", "New York")
+// Output: {name: "John", age: 30, city: "New York"}
+const testObjEx10 = { name: "John", age: 30 };
+
+
+console.log('----------------------------');
+// 11. Write a function that takes an object and a key, and removes the key-value pair from the object.
+// Input: ({name: "John", age: 30, city: "New York"}, "city")
+// Output: {name: "John", age: 30}
+const testObjEx11 = { name: "John", age: 30, city: "New York" };
+
+
+console.log('----------------------------');
+// 12. Write a function that takes two arrays of equal length and returns an object with the first array as keys and the second array as values.
+// Input: (["name", "age", "city"], ["John", 30, "New York"])
+// Output: {name: "John", age: 30, city: "New York"}
+const array1Ex12 = ["name", "age", "city"];
+const array2Ex12 = ["John", 30, "New York"];
```

Comment:
In this exercises, you can solve it by using a loop. You'll loop through first array and for each element create a key of an object and assign it's value to the value at the same position in the second array.

Make sure to check first that arrays have equal length and only then create the object, otherwise log an error message.